### PR TITLE
Editor can publish articles

### PR DESCRIPTION
**Pull request by Alex-Bellamy**
### User Story
```
As an editor,
In order to make sure that the articles have good standards
I would like only editors to be able to publish articles.
```
## Changes proposed in this pull request:
* Updating flow once signed in and facility for editor to publish articles
* Amending several class components to functional components
* Updating endpoints for fetching from backend and updating failing previous tests
## What I have learned working on this feature:
* Class to functional components
## Screenshots
Screenshot 2020-08-18 at 17 46 38

### CODE REVIEW

File: cypress/integration/editorCanPublishArticles.feature.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+describe("Editor can publish articles", () => {
+  context("successfully", () => {
```
Comment (viktorsommar) [inline cypress/integration/editorCanPublishArticles.feature.js:2]: Will there be a sad path as well?

File: cypress/integration/editorCanPublishArticles.feature.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+      cy.route({
+        method: "GET",
+        url: "http://localhost:3000/api/v1/admin/articles**",
+        response: "fixture:unpublish_articles.json",
+      });
+
+      cy.route({
+        method: "GET",
+        url: "http://localhost:3000/api/v1/articles/1",
```
Comment (viktorsommar) [inline cypress/integration/editorCanPublishArticles.feature.js:26]: What are we testing here? Should the url be /admin/articles/1?

File: cypress/integration/journalistCanSeeArticles.feature.js [hunk 3]
```diff
@@ -4,52 +4,47 @@ describe('Journalist can see list of their articles', () => {
...
     cy.route({
       method: "GET",
       url: "http://localhost:3000/api/v1/auth/**",
-      response: "fixture:registration_response.json",
-    })
+      response: "fixture:login_response.json",
+    });
+    cy.route({
+      method: "GET",
+      url: "http://localhost:3000/api/v1/articles/1",
```
Comment (viktorsommar) [inline cypress/integration/journalistCanSeeArticles.feature.js:21]: Should be /api/v1/admin/articles/1

File: src/components/ArticlePage.jsx [hunk 4]
```diff
@@ -1,94 +1,93 @@
...
+    getArticles()
+  }, [])
+
+  const getArticles = async () => {
     let response;
-    response = await axios.get("/articles")
-    this.setState({ articles: response.data.articles });
+    if (userRole === 'editor') {
+      response = await axios.get("/admin/articles", {
+        params: {
```
Comment (viktorsommar) [inline src/components/ArticlePage.jsx]: We already set the params in the backend, it is not necessary to set it in the front end

File: src/components/ArticlePage.jsx [hunk 5]
```diff
@@ -1,94 +1,93 @@
...
+    } else {
+      response = await axios.get("/articles")
+    }
+    setArticles(response.data.articles);
   };
 
-  getSingleArticle = async (event) => {
+  const getSingleArticle = async (event) => {
     let id = event.target.parentElement.dataset.id;
     let response = await axios.get(`/articles/${id}`);
```
Comment (viktorsommar) [inline src/components/ArticlePage.jsx:27]: What are we expecting to see here? Should the request be made at /admin/articles?

**Review by viktorsommar — CHANGES_REQUESTED** [review]

File: src/components/ArticlePage.jsx (hunk 5 above)
Comment (Odeane) [reply src/components/ArticlePage.jsx:27]: There is no need to make a request to the admin controller for a single article

**Review by Cma-B — APPROVED** [review]

**Review by oliverochman — APPROVED** [review]

**Pull request merged**